tests/handlers/records-write.spec.ts

```diff
@@ -478,6 +478,262 @@ export function testRecordsWriteHandler(): void {
+        });
+
+        xit('should not allow entity using a non-delegated grant as a delegated grant to invoke write', async () => {
+        });
+
+        xit('should not allow entity using a non-delegated grant as a delegated grant to invoke read', async () => {
+        });
+
+        xit('should not allow entity using a non-delegated grant as a delegated grant to invoke query', async () => {
+        });
+
+        xit('should not allow entity using a non-delegated grant as a delegated grant to invoke delete', async () => {
+        });
+
+        it('should fail if invoking a delegated grant that is issued to a different entity to write', async () => {
+          // scenario:
+          // 1. Alice creates a delegated grant for Bob
+          // 2. Carol starts a chat thread with Alice
+          // 3. Daniel stole Bob's delegated grant and attempts to write to Carol's DWN
+          // 4. Daniel's attempt should fail
+          const alice = await DidKeyResolver.generate();
+          const bob = await DidKeyResolver.generate();
+          const carol = await DidKeyResolver.generate();
+          const daniel = await DidKeyResolver.generate();
+
+          // Carol has the chat protocol installed
+          const protocolDefinition = threadRoleProtocolDefinition;
+          const protocol = threadRoleProtocolDefinition.protocol;
+          const protocolsConfig = await TestDataGenerator.generateProtocolsConfigure({
+            author: carol,
+            protocolDefinition
+          });
+          const protocolWriteReply = await dwn.processMessage(carol.did, protocolsConfig.message);
+          expect(protocolWriteReply.status.code).to.equal(202);
+
+          // Carol starts a chat thread
+          const threadRecord = await TestDataGenerator.generateRecordsWrite({
+            author       : carol,
+            protocol     : protocolDefinition.protocol,
+            protocolPath : 'thread',
+          });
+          const threadRoleReply = await dwn.processMessage(carol.did, threadRecord.message, threadRecord.dataStream);
+          expect(threadRoleReply.status.code).to.equal(202);
+
+          // Carol adds Alice as a participant in the thread
+          const participantRoleRecord = await TestDataGenerator.generateRecordsWrite({
+            author       : carol,
+            recipient    : alice.did,
+            protocol     : protocolDefinition.protocol,
+            protocolPath : 'thread/participant',
+            contextId    : threadRecord.message.contextId,
+            parentId     : threadRecord.message.recordId,
+            data         : new TextEncoder().encode('Alice is my friend'),
+          });
+          const participantRoleReply = await dwn.processMessage(carol.did, participantRoleRecord.message, participantRoleRecord.dataStream);
+          expect(participantRoleReply.status.code).to.equal(202);
+
+          // Alice creates a delegated grant for device X and device Y
```

Comment:
```suggestion
          // Alice creates a delegated grant for Bob to act as Alice.
```

Reply:
Thanks for picking it up! Yet another C&P error! Fixed.